components/datetime/src/raw/datetime.rs

```diff
@@ -297,11 +373,65 @@ impl DateTimeFormatter {
         })
     }
 
-    /// Constructor that takes a selected [`Locale`], reference to a [`DataProvider`] and
-    /// a list of options, then collects all data necessary to format date and time values into the given locale.
     #[inline(never)]
-    pub fn try_new<D>(
-        data_provider: &D,
+    #[cfg(feature = "data")]
+    pub fn try_new(
```

Comment:
Thought: It would be nice if we could avoid duplicating this code but I don't see a way to do it so I think duplicating is the least offensive option

Reply:
We did decide to have different constructor trees for recursive constructors, this is the result

Reply:
Yes I understand. We also said that when constructors needed to be duplicated, we needed to ensure good test coverage for both paths. Can you make sure the old code path is covered by sufficient tests?